exonum/src/node/basic.rs

```diff
@@ -18,31 +18,31 @@ use std::{error::Error, net::SocketAddr};
 
 use super::{NodeHandler, NodeRole, RequestData};
 use helpers::Height;
-use messages::{Any, Connect, Message, PeersRequest, RawMessage, Status};
+use messages::{Any, Message, PeerList, PeersRequest, RawMessage, Status};
+use node::ConnectInfo;
 
 impl NodeHandler {
     /// Redirects message to the corresponding `handle_...` function.
     pub fn handle_message(&mut self, raw: RawMessage) {
         match Any::from_raw(raw) {
-            Ok(Any::Connect(msg)) => self.handle_connect(msg),
             Ok(Any::Status(msg)) => self.handle_status(&msg),
             Ok(Any::Consensus(msg)) => self.handle_consensus(msg),
             Ok(Any::Request(msg)) => self.handle_request(msg),
             Ok(Any::Block(msg)) => self.handle_block(&msg),
             Ok(Any::Transaction(msg)) => self.handle_tx(&msg),
             Ok(Any::TransactionsBatch(msg)) => self.handle_txs_batch(&msg),
+            Ok(Any::PeerList(msg)) => self.handle_peers_exchange(&msg),
             Err(err) => {
                 error!("Invalid message received: {:?}", err.description());
             }
         }
     }
 
-    /// Handles the `Connected` event. Node's `Connect` message is sent as response
-    /// if received `Connect` message is correct.
-    pub fn handle_connected(&mut self, address: &SocketAddr, connect: Connect) {
-        info!("Received Connect message from peer: {:?}", address);
-        // TODO: use `ConnectInfo` instead of connect-messages. (ECR-1452)
-        self.handle_connect(connect);
+    /// Handles the incoming connection. Node connects to the sender
+    /// if received `PeerList` is correct.
+    pub fn handle_connected(&mut self, peers_exchange: &PeerList) {
+        info!("Received PeerList  from peer: {:?}", peers_exchange);
+        self.handle_peers_exchange(&peers_exchange);
```

Comment:
I think various `peers_exchange` mentions in the code should be replaced with `peer_list` to match the new struct name.